lib/internal/modules/esm/loader.js

```diff
@@ -594,7 +616,7 @@ class ESMLoader {
       shortCircuited: false,
     };
 
-    const validate = (hookErrIdentifier, nextUrl, ctx) => {
+    const validate = (hookErrIdentifier, { 0: nextUrl, 1: ctx }) => {
```

Comment:
Is`{ 0: nextUrl, 1: ctx }` to destructure an array? Could you do `(hookErrIdentifier, [ nextUrl, ctx ]) => {` instead?

Reply:
Object destructuring works fine on an array, and array destructuring is unsafe because it invokes `Array.prototype[Symbol.iterator]`

Reply:
Yes, what Jordon said